test/SmartDetectorsSharedTests/SmartDetectorLoaderTests.cs

```diff
@@ -83,8 +83,8 @@ public void TestInitialize()
 
             this.manifests = new Dictionary<string, SmartDetectorManifest>()
             {
-                ["1"] = new SmartDetectorManifest("1", "Test Smart Detector", "Test Smart Detector description", Version.Parse("1.0"), "TestSmartDetectorLibrary.dll", "TestSmartDetectorLibrary.TestSmartDetector", new List<ResourceType>() { ResourceType.Subscription }, new List<int> { 60 }, null),
-                ["2"] = new SmartDetectorManifest("2", "Test Smart Detector with dependency", "Test Smart Detector with dependency description", Version.Parse("1.0"), "TestSmartDetectorLibrary.dll", "TestSmartDetectorLibrary.TestSmartDetectorWithDependency", new List<ResourceType>() { ResourceType.Subscription }, new List<int> { 60 }, null)
+                ["1"] = new SmartDetectorManifest("1", "Test Smart Detector", "Test Smart Detector description", Version.Parse("1.0"), "TestSmartDetectorLibrary.dll", "TestSmartDetectorLibrary.TestSmartDetector", new List<ResourceType>() { ResourceType.Subscription }, new List<int> { 60 }, null, null),
```

Comment:
>null [](start = 297, length = 4)

you add null everywhere.
Consider adding a test somewhere that checks something for the code you added.

Reply:
no I didn't - check out SmartDetectorRunnerTests

---
In reply to: [233436287](https://github.com/Azure/SmartDetectorsMonitoringAppliance/pull/23#discussion_r233436287) [](ancestors = 233436287)